junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
I have added a new version with merged catch blocks. 

I also removed the catch from `awaitBarrier` and `notifyBarrier`. This might require changes in tests that are not declared to throw Exception and are using barriers. In gateway I found only 3 cases.